routes/booking.js

```diff
@@ -0,0 +1,170 @@
+require("express-async-errors");
+const express = require("express");
+const { Booking } = require("../models/booking");
+const { Room } = require("../models/room");
+const isClient = require("../middleware/isClient");
+const isManager = require("../middleware/isManager");
+const router = express.Router();
+
+router.post("/requestToBook/", isClient, async (req, res) => {
+  const { arrivalDate, departureDate, roomId } = req.query;
+  const { user } = req;
```

Comment:
Once again, to keep up consistency, I would suggest to you to keep one source of truth, and use context to get the user info.